src/vs/workbench/common/contributions.ts

```diff
@@ -126,11 +128,20 @@ class WorkbenchContributionsRegistry implements IWorkbenchContributionsRegistry
 		}
 	}
 
-	private safeCreateInstance(instantiationService: IInstantiationService, ctor: IConstructorSignature<IWorkbenchContribution>): void {
+	private safeCreateInstance(instantiationService: IInstantiationService, logService: ILogService, ctor: IConstructorSignature<IWorkbenchContribution>, phase: LifecyclePhase): void {
+		const now: number | undefined = phase < LifecyclePhase.Restored ? Date.now() : undefined;
+
 		try {
 			instantiationService.createInstance(ctor);
 		} catch (error) {
-			console.error(`Unable to instantiate workbench contribution ${ctor.name}.`, error);
+			logService.error(`Unable to instantiate workbench contribution ${ctor.name}.`, error);
+		}
+
+		if (typeof now === 'number') {
+			const time = Date.now() - now;
+			if (time > 5) {
+				logService.warn(`Workbench contribution ${ctor.name} blocked restore phase by ${time}ms.`);
```

Comment:
Careful - in the product (after minification) this is just `Ke` or `Xa` etc

Reply:
@jrieken ouch, any advice what to log then? I guess we would have to introduce identifiers to get out of it...

Reply:
Not really. You can just print the source of the ctor. I do something similar with extension language features - I know the extension but if its suggest or definitions etc. I then just print the caller: https://github.com/microsoft/vscode/blob/342394d1e7d43d3324dc2ede1d634cffd52ba159/src/vs/workbench/api/common/extHostLanguageFeatures.ts#L1891

Reply:
Ok let me see to add an identifier then.

Reply:
@jrieken can you approve again, `registerWorkbenchContribution` now carries an `id` that we can use to print it.